pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -748,3 +759,29 @@ func (t *ResolvedPipelineTask) hasResultReferences() bool {
 func isCustomRunCancelledByPipelineRunTimeout(cr *v1beta1.CustomRun) bool {
 	return cr.Spec.StatusMessage == v1beta1.CustomRunCancelledByPipelineTimeoutMsg
 }
+
+// CheckMissingResultReferences returns an error if it is missing any result references
+// validates any array result indexes to ensure the reference is in bounds.
+// Missing result references can occur if task fails to produce a result but has
+// OnError: continue (ie TestMissingResultWhenStepErrorIsIgnored)
+func CheckMissingResultReferences(pipelineRunState PipelineRunState, targets PipelineRunState) error {
```

Comment:
not a block for the PR: the docstrings says "validates any array result indexes to ensure the reference is in bounds.", is it referring to `validateArrayResultsIndex`. Can we call `validateArrayResultsIndex` inside this function?

Reply:
@Yongxuanzhang Good catch. I forgot I updated this so that validateArrayResultIndex() is called outside of this function 
so that we are able to return an error that immediately fails the pipeline run. Removed this from the doc string. https://github.com/tektoncd/pipeline/blob/b5d6047bf924006345f9599be42640fd1b758595/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L556